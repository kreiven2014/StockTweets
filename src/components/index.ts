export { default as BaseInput } from "src/components/base_input/base_input";
export { default as BaseText } from "src/components/base_text/base_text";
export { default as Button } from "src/components/button/button";
export { default as Checkbox } from "src/components/checkbox/checkbox";
export { default as FastImageComponent } from "src/components/image/image";
export { default as HeaderTitle } from "src/components/header/header_title/header_title";
export { default as HeaderRightTape } from "src/components/header/header_right_tape/header_right_tape";
export { default as KeyboardWrapper } from "src/components/keyboard_wrapper/keyboard_wrapper";
export { default as CustomWrapperKeyboard } from "src/components/keyboard_wrapper/custom_wrapper_keyboard";
export { default as Row } from "src/components/row/row";
export { default as Icons } from "src/components/icons/icons";
export { default as ImagePlaceholder } from "src/components/avatar_placeholder";
