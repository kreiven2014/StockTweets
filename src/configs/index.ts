export { default as MAIN_CONFIG } from "./main_config";
export { default as DATE_FORMATS } from "./date_formats";
export { default as REGEXP_CONFIG } from "./regexp_config";
