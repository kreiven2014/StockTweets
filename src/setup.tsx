/* REACT */
import React from "react";
import { YellowBox } from "react-native";

/* MODULES */
import { I18nextProvider } from "react-i18next";
import { ActionSheetProvider } from "@expo/react-native-action-sheet";

/* CUSTOM MODULES */
import App from "src/app";
import i18n from "src/i18n/i18n"; // initialized i18next instance

YellowBox.ignoreWarnings(["Slider has been extracted"]);

export default () => (
  <I18nextProvider i18n={i18n}>
    <ActionSheetProvider>
      <App />
    </ActionSheetProvider>
  </I18nextProvider>
);
