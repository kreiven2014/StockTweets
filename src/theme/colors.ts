export default {
  WHITE: "#FFFFFF",
  BLACK: "#000000",
  SILVER_CHALICE: "#AFAFAF",
  DENIM: "#1568BE",
  ATHENS_GRAY: "#F5F6F8",
  MANATEE: "#9091A0",
  BLUEBERRY: "#7A81F7",
  SALOMIE: "#FED994",
  CARRIBEAN_GREEN: "#02CF8E",
  BLACK_OPACITY06: "rgba(0, 0, 0, 0.6)",
  TURQUOISE: "#30D5C8",
  BLUE: "#0000FF",
  cloudBurst: "#2430521a",
  trout: "#494E5C",
  shuttleGray: "#676B7A",
  tuna: "#2F323D",
  TRANSPARENT: "transparent",
  RED: "#FF0000",
  ERROR: "#F56358",
  LINK: "#2E5FE6",
  INPUT: {
    TEXT_COLOR: "#9094A3",
    BORDER_COLOR: "#D7D9E0",
  },
  BUTTON: {
    BACKGROUND_COLOR: "#676B7A",
  },
  CHECKBOX: {
    OUTERBOX_BORDER: "#D7D9E0",
    LABEL: "#2F323D",
  },
  HEADER: {
    TEXT: "#2F323D",
  },
};
