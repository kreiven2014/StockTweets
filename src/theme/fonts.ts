/* REACT */
// import { Platform } from "react-native";

export default {
  // TODO(anybody): add some font constants if need them !
  // ArialRoundedMTBold: Platform.select({ ios: "ArialRoundedMTBold", android: "ArialRounderMTBold" }),
  // JosefinSans_SemiBoldItalic: "JosefinSans-SemiBoldItalic",
};
