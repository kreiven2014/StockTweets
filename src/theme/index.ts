export { default as FONTS } from "./fonts";
export { default as COLORS } from "./colors";
export { default as SIZES } from "./sizes";
