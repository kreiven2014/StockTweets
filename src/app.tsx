/* REACT */
import React from "react";
import { StyleSheet } from "react-native";

/* MODULES */
import { SafeAreaView } from "react-navigation";

/* CUSTOM MODULES */
import Navigator from "./navigation";

/* CUSTOM MODULES */
import { HeaderRightTape } from "src/components";

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
});

export default () => (
  <SafeAreaView
    style={styles.root}
    forceInset={{
      horizontal: "always",
    }}>
    <Navigator />
    <HeaderRightTape />
  </SafeAreaView>
);
