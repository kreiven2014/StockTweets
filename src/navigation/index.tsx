/* REACT */
import React, { Component } from "react";
import { Platform } from "react-native";

/* MODULES */
import SplashScreen from "react-native-splash-screen";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import { createDrawerNavigator } from "@react-navigation/drawer";

/* CUSTOM MODULES */
import OnBoarding from "src/containers/onboarding/onboarding";
import Home from "src/containers/home/home";
import { defaultNavigationOptions } from "src/navigation/navigation_helpers";
import { COLORS } from "src/theme";
import styles from "./styles";

export interface Props {
  isAuthorized: boolean;
}

const Stack = createStackNavigator();
const Drawer = createDrawerNavigator();

// IN CASE we need to use modal drawer for other screens
function MyDrawer() {
  return (
    <Drawer.Navigator
      initialRouteName="Home"
      drawerStyle={styles.drawerStyle}
      drawerContentOptions={{
        activeTintColor: COLORS.WHITE,
        itemStyle: { backgroundColor: COLORS.TRANSPARENT },
      }}>
      <Drawer.Screen name="Menu1" component={Home} />
      <Drawer.Screen name="Menu2" component={Home} />
      <Drawer.Screen name="Menu3" component={Home} />
    </Drawer.Navigator>
  );
}
export default class extends Component<Props> {
  static defaultProps: Props;

  componentDidMount() {
    setTimeout(SplashScreen.hide, Platform.OS === "ios" ? 500 : 0);
  }

  // ==================
  // ===== RENDER =====
  // ==================

  render(): JSX.Element {
    return (
      <NavigationContainer>
        <Stack.Navigator
          screenOptions={{
            ...defaultNavigationOptions,
            //@ts-ignore disable header for whole app
            header: () => { },
          }}
          initialRouteName="OnBoarding">
          <Stack.Screen name="OnBoarding" component={OnBoarding} />
          <Stack.Screen
            name="Home"
            component={MyDrawer}
            options={{ gestureEnabled: false }}
          />
        </Stack.Navigator>
      </NavigationContainer>
    );
  }
}
