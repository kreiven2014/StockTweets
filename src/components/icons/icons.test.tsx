/* REACT */
import React from "react";

/* MODULES */
import renderer from "react-test-renderer";

/* CUSTOM MODULES */
import Icon from "./icons";
import { COLORS } from "src/theme";

describe("Icons tests", () => {
  test("renders correctly", (): void => {
    const tree = renderer
      .create(<Icon name="credit_cart" size={12} color={COLORS.WHITE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // eslint-disable-next-line quotes
  test('renders correctly with type="material"', (): void => {
    const tree = renderer
      .create(
        <Icon type="material" name="ac-unit" size={12} color={COLORS.BLACK} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
