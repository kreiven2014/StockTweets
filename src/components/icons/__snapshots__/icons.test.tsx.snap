// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons tests renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontSize": 20.5,
      },
      undefined,
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icons tests renders correctly with type="material" 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 20.5,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
